### DISCUSSION TITLE
Api Key Visibility

### DISCUSSION THREAD

**Discussion by tracsorenson** [datalust/seq-tickets#1644] [Help]
The Api Keys that we generated in an older version of Seq are now no longer fully visible.  Only the first 4 characters are visible (FQzE·····) and we have no other record of what they are.  Is there any way to view the full key?

**Comment by liammclennan** [comment]
Hi @tracsorenson, 

For your security Seq does not store the API Key tokens so unfortunately there is no way to retrieve them from Seq. If they are in use then you might be able to get them from the clients. 

If you have a large number of API keys to recover then it might be worthwhile using the API (https://docs.datalust.co/docs/server-http-api#apiapikeys) to read the old keys and create identical ones.
